### PR TITLE
Add documentation of sun/moon orientation/scale differences

### PR DESCRIPTION
**Pull request by Desour**
- Absolute orientation, as requested in #11940.
- Scale thing found by looking around.
  The exact values were found in the code:
  https://github.com/minetest/minetest/blob/163d3547e65a6cea8a3e555557407e88d8e09183/src/client/sky.cpp#L145
  https://github.com/minetest/minetest/blob/163d3547e65a6cea8a3e555557407e88d8e09183/src/client/sky.cpp#L589
  `0.07*1.7 = 0.119`
  Similar for moon.
  Mesh sun and moon also seem to be the same if you interpret the other shine as size.

## To do

This PR is a Ready for Review.

## How to test

- Use non-symmetric moon and sun textures (ie. ruben ;-) ).
  Use `/time` to compare orientations.
- Use worldedit's `//lua` to adjust the moon's scale: `//lua minetest.get_player_by_name("singleplayer"):set_moon({scale = 0.119/0.076})`
  Look straight up and do `/time 0` and `/time 12000` to compare sizes.

### CODE REVIEW

**Comment by Zughy** [comment]
:-1: 
This only makes the documentation highly confusing. A healthier approach is to fix the size instead

**Comment by wsor4035** [comment]
in its current form, it is pretty confusing, perhaps it could be worded better?

**Comment by Desour** [comment]
Is it better now? I don't really see what else could cause confusion.

>A healthier approach is to fix the size instead

Everything that needs to be said was already said at #11902. This PR follows the discussion's results.

**Comment by rubenwardy** [comment]
Given I've never even noticed this, I doubt it would break any mods if the size was made equal

**Comment by Desour** [comment]
The size difference isn't slight.

I don't see how this is different to the rotation case.

**Comment by Wuzzy2** [comment]
No, please don't randomly change the size of the moon to make it "consistent". As said before, it's a noticable size difference, if this is changed, then the moon of all games will suddenly be much larger.

I'm in favor of this documentation. I don't think it is confusing at all.

**Comment by Wuzzy2** [comment]
Btw: `0.119/0.076 = 1.5657`, i.e. the size difference is ca. +57%. This is significant.

**Comment by Zughy** [comment]
About the size: one can be shrinked and the other made bigger, so to meet halfway. Nobody will notice

**Comment by Wuzzy2** [comment]
> Nobody will notice

Are you sure about that?

https://www.satoshiupload.com/images/xAFKpGapmt.png

**Comment by Wuzzy2** [comment]
Can we *please* stop breaking things for no good reason? You're literally proposing a breakage that is *completely* avoidable, only because you personally dislike a minor inconsistency. The breakage, however, will be real and will affect ALL games.

"I don't like that the moon is too small" is not sufficient reasoning to justify breakage. It's the game's job to find the correct moon size, Minetest should not do a rugpull.

**Comment by Zughy** [comment]
> https://www.satoshiupload.com/images/xAFKpGapmt.png

Read again my previous comment

> You're literally proposing a breakage

Slightly scaling two graphical elements you can't interact with in any way is not a breakage, please stop using that word out of turn

**Comment by Wuzzy2** [comment]
But the size of the moon might be intended tho for the games that already exist. Changing the moon size means you're pulling the rug under game designers, forcing them to update the game only make it look exactly like before. I don't like this attitude. I don't want minor updates to forcefully change the look-and-feel of games (unless it is *really* minor which this isn't).
Yes, it might be only a small thing, but these small things add up over time, only adding up to frustration.
Again, this change is *completely* avoidable and I simply don't see a sufficient reason to do it.

Oh, and I do consider this a breakage, not in a techincal sense (no crash), but in a graphical sense. It breaks with the expectation of game makers who have a reasonable expectation that their game roughly looks and feels like before after minor Minetest version updates, including the sky. Also, a size change of +57% is significant, not “slight”. I am guessing your proposal might cut that to a half, but then it would affect both sun *and* moon. That's not helping.

If this were a size change of like 1% or less for the reason of API consistency, THEN I would have no issue with it. Or if you would propose a size change for a major release like 6.0.0, THEN we could talk. But this does not seem to be the case.

Your own "compromise solution" is just a bad compromise: This is still an unilateral resize, it would *still* do a rugpull under ALL games with no fault of the game makers, which I do not want. This PR is overwriting the explicit wishes of the game designers. Worse, this affects your game *even if* you have explicitly set the moon size before. After this PR, game makers are *forced* to add code only to make it look like before again. It gets even worse if the game wants to support older 5.X.X versions. As there is no `minetest.features` update in this PR, this PR *guarantees* the game's chosen moon size will be incorrect (i.e. against game dev intentions) in at least one Minetest version.

The argument for consistency does not outweigh my concern for backwards-compability (also graphically spoken). As a game maker, *I do not care at all* about this very minor inconsistency in the API, it is trivial to deal with and is a complete non-issue. don't fix what ain't broken.

Long story short: If you have a problem with the default moon or sun size in Minetest, just *change it in your own game*. That way, everyone can be happy: You get your desired moon size, everyone else enjoys a stable Minetest. Why is not an option for you?

NOTE: The PR itself is good though, it is a no-brainer documentation update, and I fully approve of the PR. I just highly disagree with Zughy's takes here.

File: doc/lua_api.txt [hunk 1]
```diff
@@ -6946,13 +6946,16 @@ object you are working with still exists.
             (default: `true`)
         * `texture`: A regular texture for the sun. Setting to `""`
             will re-enable the mesh sun. (default: "sun.png", if it exists)
+            At sunrise, the texture appears non-rotated.
```
Comment (sfan5) [inline doc/lua_api.txt:6949]: compared to what?

File: doc/lua_api.txt [hunk 2]
```diff
@@ -6946,13 +6946,16 @@ object you are working with still exists.
...
         * `tonemap`: A 512x1 texture containing the tonemap for the sun
             (default: `"sun_tonemap.png"`)
         * `sunrise`: A regular texture for the sunrise texture.
             (default: `"sunrisebg.png"`)
         * `sunrise_visible`: Boolean for whether the sunrise texture is visible.
             (default: `true`)
         * `scale`: Float controlling the overall size of the sun. (default: `1`)
+            Note: For equal `scale` values, the sun is bigger than the moon by a
+            factor of `0.119/0.076`.
```
Comment (sfan5) [inline doc/lua_api.txt:6958]:
```suggestion
            Note: For legacy reasons, the sun is bigger than the moon by a
            factor of `0.119/0.076`.
```

File: doc/lua_api.txt (hunk 1 above)
Comment (Desour) [reply doc/lua_api.txt:6949]: Compared to the screen.
This assumes the current FPS-style camera.

File: doc/lua_api.txt (hunk 2 above)
Comment (Desour) [reply doc/lua_api.txt]: Done.
But I've kept the "for equal `scale` values" because one might otherwise read it as "the sun is always bigger than the moon by this factor. if you set the one scale, the other one is automatically adjusted to keep this fact", which is wrong. Instead, one can set both scale values independently.

File: doc/lua_api.txt (hunk 1 above)
Comment (Desour) [reply doc/lua_api.txt]: Description adjusted for clarity.

File: doc/lua_api.txt [hunk 3]
```diff
@@ -6946,13 +6946,17 @@ object you are working with still exists.
             (default: `true`)
         * `texture`: A regular texture for the sun. Setting to `""`
             will re-enable the mesh sun. (default: "sun.png", if it exists)
+            At sunrise, the texture appears non-rotated on the screen if one
+            looks at it.
```
Comment (x2048) [inline doc/lua_api.txt:6950]: Proposed wording:

The texture appears non-rotated at sunrise and rotated 180 degrees (upside down) at sunset.

File: doc/lua_api.txt [hunk 4]
```diff
@@ -6962,11 +6966,15 @@ object you are working with still exists.
             (default: `true`)
         * `texture`: A regular texture for the moon. Setting to `""`
             will re-enable the mesh moon. (default: `"moon.png"`, if it exists)
-            Note: Relative to the sun, the moon texture is rotated by 180°.
+            At sunrise, the texture appears non-rotated on the screen if one
+            looks at it.
```
Comment (x2048) [inline doc/lua_api.txt:6970]: Same proposed wording here.

The texture appears non-rotated at sunrise / moonset and rotated 180 degrees (upside down) at sunset / moonrise.

File: doc/lua_api.txt [hunk 5]
```diff
@@ -6946,13 +6946,17 @@ object you are working with still exists.
...
         * `tonemap`: A 512x1 texture containing the tonemap for the sun
             (default: `"sun_tonemap.png"`)
         * `sunrise`: A regular texture for the sunrise texture.
             (default: `"sunrisebg.png"`)
         * `sunrise_visible`: Boolean for whether the sunrise texture is visible.
             (default: `true`)
         * `scale`: Float controlling the overall size of the sun. (default: `1`)
+            Note: For legacy reasons, the sun is bigger than the moon by a factor
+            of `0.119/0.076` for equal `scale` values.
```
Comment (x2048) [inline doc/lua_api.txt:6959]: Change ratio to '1.57', because it's simpler and more informative.

**Review by x2048 — CHANGES_REQUESTED** [review]

File: doc/lua_api.txt (hunk 5 above)
Comment (Desour) [reply doc/lua_api.txt:6959]: I've written it like this so that one can find the values in the src by grepping.

File: doc/lua_api.txt (hunk 3 above)
Comment (Desour) [reply doc/lua_api.txt]: Sounds good. Done.

File: doc/lua_api.txt (hunk 5 above)
Comment (Desour) [reply doc/lua_api.txt]: Done.
I've also made comments to the src to allow grepping for the magic numbers.

**Review by sfan5 — APPROVED** [review]

**Review by x2048 — APPROVED** [review]
